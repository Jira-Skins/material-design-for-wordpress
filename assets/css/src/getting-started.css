@import "@material/typography/dist/mdc.typography.css";
@import "@material/layout-grid/dist/mdc.layout-grid.css";
@import "@material/button/dist/mdc.button.css";
@import "./gsm/index.css";
