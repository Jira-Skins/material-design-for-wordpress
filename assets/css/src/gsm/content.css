.material-gsm__content {
	padding-right: 1rem;
}
