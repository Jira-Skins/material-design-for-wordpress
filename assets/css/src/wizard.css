@import "@material/typography/dist/mdc.typography.css";
@import "@material/layout-grid/dist/mdc.layout-grid.css";
@import "@material/button/dist/mdc.button.css";
@import "./wizard/index.css";
