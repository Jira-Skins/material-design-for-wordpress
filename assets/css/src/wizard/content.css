.material-wizard__content {
	background-color: #fff;
	box-shadow: 0 2px 2px rgba(0, 0, 0, 0.15);
	padding: 2.5rem;

	& p {
		font-family: var(--mdc-typography-body1-font-family, var(--mdc-typography-font-family, Roboto,));
		font-size: var(--mdc-typography-body1-font-size, 1rem);
		line-height: var(--mdc-typography-body1-line-height);
		font-weight: var(--mdc-typography-body1-font-weight, 400);
		letter-spacing: var(--mdc-typography-body1-letter-spacing, 0.03125em);
		text-decoration: var(--mdc-typography-body1-text-decoration, inherit);
		text-transform: var(--mdc-typography-body1-text-transform, inherit);
	}

	& hr {
		margin: 2.5rem auto;
		width: 80%;
	}

	& .material-wizard__title {
		margin-top: 0;
	}

	& .material-wizard__card img {
		width: 100%;
	}
}
