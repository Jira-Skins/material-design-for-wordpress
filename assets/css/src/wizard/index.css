@import "./base.css";
@import "./progress.css";
@import "./navigation.css";
