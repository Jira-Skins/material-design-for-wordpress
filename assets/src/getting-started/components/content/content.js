export * from './wizard';
export * from './theme';
export * from './demo';
export * from './editor';
export * from './blocks';
