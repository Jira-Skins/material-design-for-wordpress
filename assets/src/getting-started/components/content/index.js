const Content = () => {
	return (
		<div className="material-gsm__content mdc-layout-grid__cell mdc-layout-grid__cell--span-9">
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				Onboarding Wizard
			</h2>
			<p className="material-gsm__content-description">
				Curabitur mauris erat congue posuere lacus torquent et convallis,
				parturient rutrum cras senectus augue penatibus malesuada rhoncus,
				ultricies finibus pretium arcu sagittis aliquet nisi.
			</p>
			<div className="material-gsm__content-actions"></div>
		</div>
	);
};

export default Content;
