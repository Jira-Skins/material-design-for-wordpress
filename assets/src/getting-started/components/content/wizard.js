/* global mtbGsm */
import { Fragment } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import Button from '../../../wizard/components/navigation/button';

export const Wizard = props => {
	return (
		<Fragment>
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Quick Start', 'material-theme-builder' ) }
			</h2>
			<p className="material-gsm__content-description">
				Aenean egestas, ante vitae placerat tempor, felis ipsum finibus lectus,
				at eleifend neque tellus ac elit. Praesent cursus lectus felis, a ornare
				metus varius in.
			</p>
			<div className="material-gsm__content-actions">
				<Button
					style="mdc-button--raised"
					text={ __( 'Re-run Wizard', 'material-theme-builder' ) }
					trailingIcon="navigate_next"
					onClick={ props.handleClick }
					link={ mtbGsm.wizardUrl }
				/>
			</div>
		</Fragment>
	);
};
