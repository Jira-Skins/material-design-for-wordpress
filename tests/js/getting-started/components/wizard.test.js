/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Wizard } from '../../../../assets/src/getting-started/components/content/wizard';
import { TabProvider } from '../../../../assets/src/getting-started/context';

const setup = () => {
	return render(
		<TabProvider>
			<Wizard />
		</TabProvider>
	);
};

describe( 'GSM: Wizard', () => {
	beforeAll( () => {
		global.mtbGsm = {
			wizardUrl: 'http://example.com',
		};
	} );

	afterEach( cleanup );

	it( 'matches snapshot', () => {
		const wrapper = setup();
		expect( wrapper ).toMatchSnapshot();
	} );
} );
