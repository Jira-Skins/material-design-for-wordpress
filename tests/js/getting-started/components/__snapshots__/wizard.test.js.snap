// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSM: Wizard matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="material-gsm__content-title mdc-typography--headline6"
      >
        Quick Start
      </h2>
      <p
        class="material-gsm__content-description"
      >
        Aenean egestas, ante vitae placerat tempor, felis ipsum finibus lectus, at eleifend neque tellus ac elit. Praesent cursus lectus felis, a ornare metus varius in.
      </p>
      <div
        class="material-gsm__content-actions"
      >
        <a
          class="mdc-button mdc-button--raised"
          href="http://example.com"
        >
          <span
            class="mdc-button__label"
          >
            Re-run Wizard
          </span>
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon trailing-icon"
          >
            navigate_next
          </i>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      class="material-gsm__content-title mdc-typography--headline6"
    >
      Quick Start
    </h2>
    <p
      class="material-gsm__content-description"
    >
      Aenean egestas, ante vitae placerat tempor, felis ipsum finibus lectus, at eleifend neque tellus ac elit. Praesent cursus lectus felis, a ornare metus varius in.
    </p>
    <div
      class="material-gsm__content-actions"
    >
      <a
        class="mdc-button mdc-button--raised"
        href="http://example.com"
      >
        <span
          class="mdc-button__label"
        >
          Re-run Wizard
        </span>
        <i
          aria-hidden="true"
          class="material-icons mdc-button__icon trailing-icon"
        >
          navigate_next
        </i>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
